pkg/tuned/cmd/render/render.go

```diff
@@ -0,0 +1,264 @@
+	)
+}
+
+func render(inputDir []string, outputDir string) error {
+	klog.Info("Rendering files from: ", inputDir)
+	klog.Info("Rendering files into: ", outputDir)
+
+	bootstrapSafeEnv := os.Getenv("CLUSTER_NODE_TUNED_BOOTSTRAP_SAFE_ENV")
+	if len(bootstrapSafeEnv) == 0 {
+		return fmt.Errorf("Should only be run on bootstrap safe environment. Please define env var 'CLUSTER_NODE_TUNED_BOOTSTRAP_SAFE_ENV' ")
+	}
+
+	// Get pools, mConfigs and profile from inputDir
+	// Read asset directory fileInfo
+	filePaths, err := util.ListFilesFromMultiplePaths(inputDir)
+	if err != nil {
+		return fmt.Errorf("error while listing files: %w", err)
+	}
+	klog.Infof("listed files: %v", filePaths)
+	// Make output dir if not present
+	err = os.MkdirAll(outputDir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("Error while creating outputdir %s : %w", outputDir, err)
+	}
+
+	var (
+		perfProfiles []*performancev2.PerformanceProfile
+		mcPools      []*mcfgv1.MachineConfigPool
+		mcConfigs    []*mcfgv1.MachineConfig
+		tuneD        []*tunedv1.Tuned
+	)
+
+	// Iterate through the file paths and read in desired files
+	klog.Info("Iterating over listed files ... ")
+	for _, path := range filePaths {
+		file, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("error opening %s: %w", file.Name(), err)
+		}
+		defer file.Close()
+
+		manifests, err := util.ParseManifests(file.Name(), file)
+		if err != nil {
+			return fmt.Errorf("error parsing manifests from %s: %w", file.Name(), err)
+		}
+
+		// Decode manifest files
+		klog.V(4).Infof("decoding manifests for file %s...", path)
+		for idx, m := range manifests {
+			obji, err := runtime.Decode(runtimeDecoder, m.Raw)
+			if err != nil {
+				if runtime.IsNotRegisteredError(err) {
+					klog.V(4).Infof("skipping path %q [%d] manifest because it is not part of expected api group: %v", file.Name(), idx+1, err)
+					continue
+				}
+				return fmt.Errorf("error parsing %q [%d] manifest: %w", file.Name(), idx+1, err)
+			}
+
+			switch obj := obji.(type) {
+			case *performancev2.PerformanceProfile:
+				perfProfiles = append(perfProfiles, obj)
+			case *mcfgv1.MachineConfigPool:
+				mcPools = append(mcPools, obj)
+			case *mcfgv1.MachineConfig:
+				mcConfigs = append(mcConfigs, obj)
+			case *tunedv1.Tuned:
+				tuneD = append(tuneD, obj)
+			default:
+				klog.Infof("skipping %q [%d] manifest because of unhandled %T", file.Name(), idx+1, obji)
+			}
+		}
+	}
+
+	perfProfilesTuneDProfiles := make(map[*performancev2.PerformanceProfile]*tunedv1.Tuned, len(perfProfiles))
+	for _, profile := range perfProfiles {
+		tunedFromPP, err := tuned.NewNodePerformance(profile)
+		if err != nil {
+			klog.Errorf("Unable to get tuned. error : %v", err)
+			return fmt.Errorf("unable to get tuned from profile: %w", err)
+		}
+		perfProfilesTuneDProfiles[profile] = tunedFromPP
+		// add tuned from PP to the list
+		tuneD = append(tuneD, tunedFromPP)
+	}
+
+	tuneDrecommended := operator.TunedRecommend(tuneD)
```

Comment:
I wonder if this should go to an MCP loop in case there are multiple pools with multiple profiles.

Reply:
So ... still need to load all the PerformanceProfiles? or just those which will match with MCP master?

Reply:
I think master only and only on SNO.